mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoInsertMethodGenerator.java

```diff
@@ -83,18 +82,20 @@ public DaoInsertMethodGenerator(
     }
 
     // Validate the return type:
-    ReturnType returnType = parseReturnType(methodElement.getReturnType());
-    if (!SUPPORTED_RETURN_TYPES.contains(returnType.kind)) {
+    DaoReturnType returnType = parseReturnType(methodElement.getReturnType());
+    Set<DaoReturnTypeKind> supportedReturnTypes = getSupportedReturnTypes();
+    if (!supportedReturnTypes.contains(returnType.getKind())) {
       context
           .getMessager()
           .error(
               methodElement,
-              "Invalid return type: %s methods must return either void or the entity class "
```

Comment:
If we are touching that logic maybe we could move it as a method to `DaoBase` and call with `annotation` and `supportedReturnTypes` parameters from every specific method generator?

Reply:
Yes I thought about it. It's not used by every subclass but it could be worth it, I'll give it a try.